contracts/service-provider-directory/src/test_helpers/signing.rs

```diff
@@ -0,0 +1,41 @@
+use cosmwasm_std::{Addr, Coin, Deps};
+use nym_contracts_common::signing::{
+    MessageSignature, SignableMessage, SigningAlgorithm, SigningPurpose,
+};
+use nym_crypto::asymmetric::identity;
+use nym_service_provider_directory_common::{
+    signing_types::{
+        construct_service_provider_announce_sign_payload, SignableServiceProviderAnnounceMsg,
+    },
+    ServiceDetails,
+};
+use serde::Serialize;
+
+use crate::state;
+
+pub fn service_provider_announce_sign_payload(
+    deps: Deps<'_>,
+    owner: &str,
+    service: ServiceDetails,
+    deposit: Coin,
+) -> SignableServiceProviderAnnounceMsg {
+    let owner = Addr::unchecked(owner);
+    let nonce = state::get_signing_nonce(deps.storage, owner.clone()).unwrap();
+    construct_service_provider_announce_sign_payload(nonce, owner, deposit, service)
+}
+
+pub fn ed25519_sign_message<T: Serialize + SigningPurpose>(
```

Comment:
yeah, I think we should definitely have some common crate...

Reply:
actually, don't we already have `common/cosmwasm-smart-contracts/contracts-common/src/signing`? i.e. where we define `SigningPurpose`  trait.  Maybe we could put those things there?